### PR TITLE
Node pool operations should retry if they encountered quota error.

### PR DESCRIPTION
**Pull request by cindyzhengjw**
This change is to fix https://github.com/hashicorp/terraform-provider-google/issues/15579. Node pool create, udpate and delete should treat the concurrent operation quota error as a retriable error. 

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
container: fixed concurrent ops' quota-error to be retriable in `google_container_node_pool `
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Hi Cindy, I notice this PR is in draft. Let me know when you are ready for me to make a pass

**Comment by cindyzhengjw** [comment]
Hi @ScottSuarez , thanks for the reminding! The PR is now sent our for review. BTW I noticed there are some failures with the checks, and I also encountered some problems with `make test` and `generated providers` (they're mentioned in the PR description) I'm not super familiar with Terraform and the contribute process, could you please take a look? Thank you so much!

**Comment by ScottSuarez** [comment]
Looks like there are some linting failures downstream
https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/6041170707/job/16393533033?pr=8828

Could you correct these? You can see the checks at the bottom of this PR

File: mmv1/third_party/terraform/tpgresource/utils.go [hunk 1]
```diff
@@ -501,14 +515,14 @@ func CheckGoogleIamPolicy(value string) error {
...
+// Retries an operation while the canonical error code is FAILED_PRECONDTION or RESOURCE_EXHAUSTED
 // which indicates there is an incompatible operation already running on the
-// cluster. This error can be safely retried until the incompatible operation
+// cluster or there are the number of allowed concurrent operations running on the cluster. These errors can be safely retried until the incompatible operation
 // completes, and the newly requested operation can begin.
 func RetryWhileIncompatibleOperation(timeout time.Duration, lockKey string, f func() error) error {
 	return resource.Retry(timeout, func() *resource.RetryError {
 		if err := transport_tpg.LockedCall(lockKey, f); err != nil {
-			if IsFailedPreconditionError(err) {
+			if IsFailedPreconditionError(err) || IsQuotaError(err) {
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tpgresource/utils.go:525]: Could this function be pulled into the cluster package? It doesn't seem to be shared by any other products.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

**Comment by ScottSuarez** [comment]
> Hi @ScottSuarez , thanks for the reminding! The PR is now sent our for review. BTW I noticed there are some failures with the checks, and I also encountered some problems with `make test` and `generated providers` (they're mentioned in the PR description) I'm not super familiar with Terraform and the contribute process, could you please take a look? Thank you so much!

Ah Just saw your comment. Sure ! So these are linting failures downstream. Are you free tomorrow? we can set 20 minutes aside and I can help you out.

**Comment by ScottSuarez** [comment]
Looks like failing unit and build tests. Anything I can help with?

**Comment by cindyzhengjw** [comment]
Thank you, Scott, for your help! The PR has successful [make test and make lint](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) run. And also verified related acceptance test by `make testacc TEST=./google/services/container TESTARGS='-run=TestAccContainerNodePool'`. There is one failure that seems related `TestAccContainerNodePool_withUpgradeSettings` but it turns out that the failure is pre-existing and should no longer be relevent.

File: mmv1/third_party/terraform/tpgresource/utils.go (hunk 1 above)
Comment (cindyzhengjw) [reply mmv1/third_party/terraform/tpgresource/utils.go]: Sure! moved to the clusters package.

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**